Oozie upload task

nntnag17 opened the pull request:
JIRAs fixed: 
LIHADOOP-13353 Hadoop Plugin oozieUpload task should upload to versioned directory.
LIHADOOP-13457 Fix Stackoverflow exception thrown by OozieUploadTask.

convexquad commented on hadoop-plugin/src/main/groovy/com/linkedin/gradle/oozie/OozieUploadTask.groovy:
```diff
@@ -21,11 +21,14 @@ class OozieUploadTask extends DefaultTask {
     try {
       logger.info("Project path: ${projectPath.toString()}");
       logger.info("Directory path: ${directoryPath.toString()}");
+      if(fs.exists(projectPath)) {
```
Ah, we need to be able to oozieUpload multiple times and have it overwrite the previous version. At least that's what I think right now - I'm not completely sure yet.  Can you cause it to overwrite? I think we want to keep the exists() method you added, even if we end up not using it.

convexquad commented on li-hadoop-plugin/src/main/groovy/com/linkedin/gradle/liutil/LiKerberosUtil.groovy:
```diff
@@ -8,12 +8,13 @@ class LiKerberosUtil {
    * We need to copy the oozie-krb5.conf file to a temporary location since we cannot point property java.security.krb5.conf to the file inside jar.
    */
   static void writeConfToTemp() {
-    logger.debug("writing oozie-krb5.conf file to ${System.getProperty("java.io.tmpdir")}");
```
Is there something in particular that made you have to change this? I think it's good to get the temp dir at the time the method is used, rather than as a static class variable, even if this means you might call System.getProperty("java.io.tmpdir") more than once. Generally, getting things like temp directory locations at static classloading times is kind of a bad thing.

convexquad commented:
I commented on a couple of issues - let's resolve them, and then I'll +1 this.

nntnag17 commented:
I've updated the pull request addressing your comments. The copyFromLocalFile method takes a boolean 'overwrite' to overwrite files but it doesn't work while copying directories. Therefore as a work around to this, I'm first deleting the hdfs directory if it exists and then calling `mkdir` and 'copyFromLocalfile`

convexquad commented on hadoop-plugin/src/main/groovy/com/linkedin/gradle/hdfs/HdfsFileSystem.groovy:
```diff
@@ -192,4 +192,30 @@ class HdfsFileSystem {
+   * @return true if p exists else returns false
+   */
+  public Boolean exists(Path p) {
+    fs.exists(p);
```
Can you add the "return" keyword to these methods, even though it's not required in Groovy?
Please make these return little boolean instead of Boolean

convexquad commented:
I added one more comment. +1 for you to merge this PR after fixing that comment.

nntnag17 commented:
I've fixed it. Merging the PR.
